src/components/DataProvider/index.js

```diff
@@ -453,7 +453,9 @@ export default class DataProvider extends Component {
         [id]: { ...loaderConfig },
       };
     }
-    this.setState(stateUpdates);
+    this.setState(stateUpdates, () => {
+      this.props.onFetchData();
```

Comment:
Would be cleaner, if `onFetchData` will be null as a default and will be called only if has been passed down